src/Symfony/Component/DependencyInjection/Attribute/AsDecorator.php

```diff
@@ -13,12 +13,27 @@
 
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
+/**
+ * An attribute to declare a decorating service.
+ */
 #[\Attribute(\Attribute::TARGET_CLASS)]
 class AsDecorator
 {
     public function __construct(
+        /**
+         * The service id to decorate.
+         */
         public string $decorates,
+
+        /**
+         * The priority of this decoration when multiple decorators are declared for the same service.
+         */
         public int $priority = 0,
+
+        /**
+         * The behavior to adopt when the decoration is invalid.Must be one of
+         * the {@see ContainerInterface} constants.
+         */
```

Comment:
We can help IDE autocompletion and static analysis with this annotation
```suggestion
         * @var ContainerInterface::*
         */
```

Reply:
Thanks, added here and in `WithLogLevel`!